### DISCUSSION TITLE
How do I go about replicating a request to subscribe to a model?

### DISCUSSION THREAD

**Discussion by NytroDev** [UltimaHoarder/UltimaScraper#1474] [Q&A]
I'm replicating all the data from onlyfans original request via **`api2/v2/payments/pay`**

However It keeps returning a code 400 with the response 

`{"error":{"code":0,"message":"PaymentType is required"},"errors":{"paymentType":["PaymentType is required","PaymentType contains invalid value"]}}`

payload: 
  paymentType : subscribe
  userId : {modelID}
  subscribeSource : profile
  amount : {subPrice}
  token : null
  unavailablePaymentGates[] : null
  
Any help with this?
  
Example, my failed request vs. one made via OnlyFans
[Failed Request](https://i.gyazo.com/24a2b63e4e01fe04066f3fbc00d3e62b.png)
[OnlyFans Request](https://i.gyazo.com/af3a6b935dce460028096e1347668688.png)
 
 
All headers are the same that I use for every OnlyFans request, which work fine apart from payment requests

- x-bc
- accept
- sess token
- app-token
- user-agent
- time 
- sign (updated to recent format via 'https://github.com/DATAHOARDERS/dynamic-rules/blob/main/onlyfans.json')

**Comment by UltimaHoarder** [comment]
Which language are you coding in?

**Reply by NytroDev** [reply]
I'm currently working with C# I can send a snippet of my code if it makes it easier.

**Reply by UltimaHoarder** [reply]
I've already got a buy_subscription function in my code. It's in the "create_user" file. You could try build it one to one with that.

**Reply by NytroDev** [reply]
I saw that, however its essentially the same thing i'm using with my snippet of code 

[Code](https://i.gyazo.com/3676741622c1fabc36b72536392ae3d5.png)

**Reply by UltimaHoarder** [reply]
> I saw that, however its essentially the same thing i'm using with my snippet of code
> 
> [Code](https://i.gyazo.com/3676741622c1fabc36b72536392ae3d5.png)

In the headers, set content-type to "application/json"

Whenever OF requests JSON data, you'll need to use that
